lib/qam.pm

```diff
@@ -73,31 +73,42 @@ sub is_patch_needed {
     }
 }
 
+sub add_repo_if_not_present {
+    my ($url, $name) = @_;
+    my $gpg = get_var('BUILD') =~ m/^MR:/ ? "-G" : "";
+    my $system_repos = script_output('zypper lr -u');
+    unless ($system_repos =~ /$url/) {
```

Comment:
I think this regex check could easily break due to unescaped special characters in `$url`. It'd be safer to call `$repolist = zypper_repos('-u')` and then `grep { $_->{uri} eq $url } @$repolist`

Reply:
Done, I don't know, but I don't think there should be any special characters in this urls, but now we will never find out. :wink: